### ISSUE TITLE
Pagination Buttons are missing `role="button"`

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2043]
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/trimble-oss/modus-web-components/blob/main/CONTRIBUTING.md)

### Describe the issue

Pagintation buttons are not accessible. the name of the link/button (previous / next) is not announced as the lasy items do not have a role. I expect the role to be button.

(tested with NVDA on Windows 11 Pro)

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Safari, Microsoft Edge

### What is the issue regarding ?

@trimble-oss/modus-web-components

### What version of npm package are you using ?

v.0.12.0

### Priority

High

### What product/project are you using Modus Components for ?

Modus

### What is your team/division name ?

Modus

### Are you willing to contribute ?

Maybe

### Are you using Modus Web Components in production ?

https://modus-web-components.trimble.com/

**Comment by coliff** [comment]
The fix for this isn't simple. The Pagination component is a list (which is correct), but within the list should be buttons. We can't simply change the next and previous to buttons because they are descendants of the list role and are currently listitems.

**Comment by prashanthr6383** [comment]
Hi @cjwinsor In material-UI, buttons are used inside lists. So should I follow MUI or should I use `role = button` on the lists?
https://mui.com/material-ui/react-pagination/

**Comment by coliff** [comment]
@prashanthr6383 - yes, buttons should be within the lists. Ideally shouldn't use `role="button"` at all - just use a `<button>` instead within the list instead.
